velox/substrait/SubstraitToVeloxPlan.cpp

```diff
@@ -60,8 +60,63 @@ core::SortOrder toSortOrder(const ::substrait::SortField& sortField) {
   }
 }
 
+/// Holds the information required to create
+/// a project node to simulate the emit
+/// behavior in Substrait.
+struct EmitInfo {
+  std::vector<core::TypedExprPtr> expressions;
+  std::vector<std::string> projectNames;
+};
+
+/// Helper function to extract the attributes required to create a ProjectNode
+/// used for interpretting Substrait Emit.
+EmitInfo getEmitInfo(
+    const ::substrait::RelCommon& relCommon,
+    const core::PlanNodePtr& node) {
+  const auto& emit = relCommon.emit();
+  int emitSize = emit.output_mapping_size();
+  EmitInfo emitInfo;
+  emitInfo.projectNames.reserve(emitSize);
+  emitInfo.expressions.reserve(emitSize);
+  const auto& outputType = node->outputType();
+  for (int i = 0; i < emitSize; i++) {
+    int32_t mapId = emit.output_mapping(i);
+    emitInfo.projectNames[i] = outputType->nameOf(mapId);
+    emitInfo.expressions[i] = std::make_shared<core::FieldAccessTypedExpr>(
+        outputType->childAt(mapId), outputType->nameOf(mapId));
+  }
+  return emitInfo;
+}
+
 } // namespace
 
+void ValidateEmitNode(const core::PlanNodePtr& planNode) {
+  if (std::dynamic_pointer_cast<const core::ValuesNode>(planNode) ||
+      std::dynamic_pointer_cast<const core::TableScanNode>(planNode)) {
+    VELOX_FAIL("processing Emit not allowed for ValuesNode and TableScanNode");
+  }
+}
```

Comment:
@mbasmanova what do you think about this validation?

Reply:
I expect this check to happen outside of processEmit call, e.g. when converting Values and Read relations we should check that these do not specify "common", e.g. we should reject plans that have emit specified for values or table scan.

Reply:
Ah I see. I will add that change.

Reply:
@mbasmanova basically looking at the code, the `TableScanNode` and `ValuesNode` are used only in the `ReadRel` associated `toVeloxPlan` methods. I think just adding a validation to 

```
core::PlanNodePtr SubstraitVeloxPlanConverter::toVeloxPlan(
    const ::substrait::ReadRel& readRel,
    std::shared_ptr<SplitInfo>& splitInfo)
```
would be enough.

Reply:
Yes, that should be sufficient.

Reply:
Added the change. Could you please take a look.